napari/layers/base/base.py

```diff
@@ -1210,7 +1210,11 @@ def _update_draw(self, scale_factor, corner_pixels, shape_threshold):
 
         self.scale_factor = scale_factor
 
-        # Round and clip data corners
+        # Sort, because negative affines can lead to
+        # top-left and bottom-right corners not actually
+        # being top-left and bottom-right
+        data_corners = np.sort(data_corners, axis=0)
```

Comment:
To understand what was going on here, I had to look up the shape of `data_corners` (2xD) so it might be a good idea to add that information to this implementation comment.